src/Params/PADnoteParameters.cpp

```diff
@@ -95,9 +103,16 @@ static const rtosc::Ports realtime_ports =
             int n = atoi(mm);
             p->setCurSampleSize(n, rtosc_argument(m,0).i);
             p->setCurBaseFreq(n, rtosc_argument(m,1).f);
-            p->sample[n].smp      = *(float**)rtosc_argument(m,2).b.data;
 
-            //XXX TODO memory management (deallocation of smp buffer)
+            rtosc_blob_t blob = rtosc_argument(m,2).b;
+            assert(static_cast<std::size_t>(blob.len) / sizeof(float*) == p->sample[n].smp.size());
+            // for each oscilgen basefunc parameter...
+            for(std::size_t idx = 0; idx < p->sample[n].smp.size(); ++idx)
+            {
+                // ...take sample buffer
+                p->sample[n].smp[idx] = ((float**)blob.data)[idx];
+            }
+            //XXX TODO memory management (deallocation of current smp buffer)
```

Comment:
I would recommend against large copies on the realtime side. Just take the pointer and send the old one back to be freed.